pkg/kubelet/container_manager_linux.go

```diff
@@ -70,23 +71,90 @@ func newSystemContainer(containerName string) *systemContainer {
 	}
 }
 
+type nodeConfig struct {
+	dockerDaemonContainerName string
+	systemContainerName       string
+	kubeletContainerName      string
+}
+
 type containerManagerImpl struct {
+	cadvisorInterface cadvisor.Interface
+	mountUtil         mount.Interface
+	nodeConfig
 	// External containers being managed.
 	systemContainers []*systemContainer
 }
 
 var _ containerManager = &containerManagerImpl{}
 
+// checks if the required cgroups subsystems are mounted.
+// As of now, only 'cpu' and 'memory' are required.
+func validateSystemRequirements(mountUtil mount.Interface) error {
+	const (
+		cgroupMountType = "cgroup"
+		localErr        = "system validation failed"
+	)
+	mountPoints, err := mountUtil.List()
+	if err != nil {
+		return fmt.Errorf("%s - %v", localErr, err)
+	}
+	expectedCgroups := map[string]bool{
+		"memory":  false,
+		"cpu":     false,
+		"cpuacct": false,
+		"cpuset":  false,
+	}
+	for _, mountPoint := range mountPoints {
+		if mountPoint.Type == cgroupMountType {
+			for _, opt := range mountPoint.Opts {
+				if _, exists := expectedCgroups[opt]; exists {
+					expectedCgroups[opt] = true
+				}
+			}
+		}
+	}
+	for subsystem, found := range expectedCgroups {
+		if !found {
+			return fmt.Errorf("%s - Cgroup subsytem %s not mounted", localErr, subsystem)
+		}
+	}
```

Comment:
nit: might worth returning all subsystems that are not mounted.

typo: s/subsytem/subsystem